modules/python/python-logtemplate-options.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2018 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "python-logtemplate-options.h"
+
+static PyTypeObject py_log_template_options_type;
+
+PyObject *
+py_log_template_options_new(LogTemplateOptions *template_options)
+{
+  PyLogTemplateOptions *self = PyObject_New(PyLogTemplateOptions, &py_log_template_options_type);
+  if (!self)
+    return NULL;
+
+  self->template_options = template_options;
+
+  return (PyObject *) self;
+}
+
+
+static PyTypeObject py_log_template_options_type =
+{
+  PyVarObject_HEAD_INIT(&PyType_Type, 0)
+  .tp_name = "LogTemplateOptions",
+  .tp_basicsize = sizeof(PyLogTemplateOptions),
+  .tp_dealloc = (destructor) PyObject_Del,
+  .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+  .tp_doc = "LogTemplateOptions class encapsulating a syslog-ng LogTemplateOptions",
+  .tp_new = PyType_GenericNew,
```

Comment:
Thanks I agree and fix it.

In the future, we might want to allow to create LogTemplateOptions, and then I will add the correct tp_new vfunction. Btw users still cannot instantiate it even with PyType_GenericNew, because the class object is not added to any namespace, so the name itself is not available. Though still better use NULL.